app/controllers/api/v1/cars_controller.rb

```diff
@@ -0,0 +1,44 @@
+class Api::V1::CarsController < ApplicationController
+  def index
+    @cars = Car.all
+    render json: @cars
+  end
+
+  def show
+    @car = Car.find(params[:id])
+    if @car
+      render json: @car, status: :ok
+    else
+      render json: { error: 'Car not found' }, status: :not_found
+    end
+  end
+
+  def create
+    @car = Car.new(car_params)
+    if @car.save
+      render json: @car, status: :created
+    else
+      render json: { error: 'Car data could not be created!!' }, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @car.update(car_params)
+      render json: @car, status: :updated, location: @car
+    else
+      render json: @car.errors, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @car = Car.find(params[:id])
+    @car.update(is_removed: true)
+  end
```

Comment:
- [ ] What happens here if the car is not successfully destroyed?  Kindly, add some error handling just like you did in the update method (line 25). 👍